subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy

```diff
@@ -537,4 +545,36 @@ apply from:'scriptPlugin.gradle'
         executedTasks == [':createFile']
         skippedTasks.empty  // The equals implementation for custom type always return false
     }
+
+    // Helper function that generate the FooType definition that can be used in the exact same way
```

Comment:
These methods could be pulled into a fixture class to make the test more readable and less long.